src/common/io.ts

```diff
@@ -124,6 +125,37 @@ export class IOReadable implements AoIORead {
       tags: [{ name: 'Action', value: 'Info' }],
     });
   }
+
+  async getEpochSettings(params?: EpochInput): Promise<AoEpochSettings> {
+    const allTags = [
+      { name: 'Action', value: 'Epoch-Settings' },
+      {
+        name: 'Timestamp',
+        value:
+          (params as { timestamp?: number })?.timestamp?.toString() ??
+          (
+            await this.arweave.blocks.getCurrent().catch(() => {
+              return { timestamp: Date.now() }; // fallback to current time
+            })
+          ).timestamp.toString(),
+      },
+      {
+        name: 'Epoch-Index',
+        value: (params as { epochIndex?: number })?.epochIndex?.toString(),
+      },
+    ];
+
+    const prunedTags: { name: string; value: string }[] = allTags.filter(
+      (tag: {
+        name: string;
+        value: string | undefined;
+      }): tag is { name: string; value: string } => tag.value !== undefined,
+    );
+
+    return this.process.read<AoEpochSettings>({
```

Comment:
Wondering if process.read should auto-prune tags, think we are repeating this frequently yeah?